lib/lightning_web/live/workflow_live/edit.ex

```diff
@@ -1049,6 +1061,10 @@ defmodule LightningWeb.WorkflowLive.Edit do
     |> Enum.any?()
   end
 
+  defp has_steps?(job) do
+    !Enum.empty?(job.steps)
+  end
```

Comment:
Just a tip with these kind of comparisons, when it comes to Ecto - if the relationship is not loaded we'll get a pattern match error here; because `Ecto.Association.NotLoaded` doesn't have the Enumerable protocol on it. So we're making an assumption here that the caller has loaded everything correctly.

If we were to abstract these kinds of functions (see comment up from this), we would use pattern matching to handle these cases.

Reply:
Yes you are right. For now the way I am making sure this a no problem is that I preload the steps anytime we load the jobs. See here:
- https://github.com/OpenFn/lightning/blob/fe313af42bf763083b723b6f7a7d26671c1d49fc/lib/lightning_web/live/workflow_live/edit.ex#L648
- https://github.com/OpenFn/lightning/blob/fe313af42bf763083b723b6f7a7d26671c1d49fc/lib/lightning_web/live/workflow_live/edit.ex#L1266

This might not be the sexiest way to do it (and I am happy to change it by pattern matching the `Ecto.Association.NotLoaded`) but it might be the best thing to do for now, given that all of this is temporary solution. Let me know what you think.